src/kudu/consensus/consensus_meta.cc

```diff
@@ -81,6 +81,23 @@ void ConsensusMetadata::clear_voted_for() {
   pb_.clear_voted_for();
 }
 
+void ConsensusMetadata::populate_previous_vote_history(
+    const PreviousVotePB& prev_vote) {
+  previous_vote_history_.emplace(prev_vote.election_term(), prev_vote);
+
+  // Prune vote history, if necessary. It is done only when the vote history
+  // size goes beyond a certain limit and the oldest term being pruned is
+  // older than the last known leader by this node.
+  const std::map<int64_t, PreviousVotePB>::const_iterator it =
+      previous_vote_history_.begin();
+  if (previous_vote_history_.size() > VOTE_HISTORY_MAX_SIZE &&
+      last_known_leader_.election_term() > it->first) {
+    last_pruned_term_ = it->first;
```

Comment:
its more intuitive to find the last_known_leader_.election_term() and eliminate everything below it,
instead of elminating only the first value

Reply:
it also gives a very good invariant that if an instance returns last_known_leader it wont be sharing voting history below that term, because it is immaterial to the vote gatherer

Reply:
I think the check for vote_history_max_size should only be used in the case that a last_known_leader_ based trimming would still lead to a large previousvotehistory...
i.e. always do the right thing which is trim below last_known_leader

Reply:
Updated as per our discussion on the last commit.